tests/foreman/cli/test_computeresource_azurerm.py

```diff
@@ -0,0 +1,629 @@
+            5. Delete created image
+
+        :expectedresults: Cloud_init image should be created, list, updated and deleted in
+                          AzureRm CR along with username
+
+        :CaseImportance: Critical
+
+        :CaseLevel: Integration
+        """
+
+        # Create
+        img_name = gen_string('alpha')
+        username = gen_string('alpha')
+        img_ft = ComputeResource.image_create(
+            {
+                'name': img_name,
+                'operatingsystem-id': module_os.id,
+                'architecture-id': module_architecture.id,
+                'uuid': AZURERM_RHEL7_UD_IMG_URN,
+                'compute-resource': module_azurerm_cr.name,
+                'username': username,
+                'user-data': 'yes',
+            }
+        )[0]
+        assert img_ft['message'] == 'Image created.'
+        assert img_ft['name'] == img_name
+
+        # Info Image
+        img_info = ComputeResource.image_info(
+            {'name': img_name, 'compute-resource': module_azurerm_cr.name}
+        )[0]
+        assert img_info['operating-system'] == module_os.title
+        assert img_info['username'] == username
+        assert img_info['uuid'] == AZURERM_RHEL7_UD_IMG_URN
+        assert img_info['user-data'] == 'true'
+        assert img_info['architecture'] == module_architecture.name
+
+        # List image
+        list_img = ComputeResource.image_list({'compute-resource': module_azurerm_cr.name})
+        assert len(list_img) == 1
+        assert list_img[0]['name'] == img_name
+
+        # Update image
+        new_img_name = gen_string('alpha')
+        new_username = gen_string('alpha')
+        result = ComputeResource.image_update(
+            {
+                'name': img_name,
+                'compute-resource': module_azurerm_cr.name,
+                'new-name': new_img_name,
+                'username': new_username,
+            }
+        )[0]
+        assert result['message'] == 'Image updated.'
+        assert result['name'] == new_img_name
+
+        img_info = ComputeResource.image_info(
+            {'name': new_img_name, 'compute-resource': module_azurerm_cr.name}
+        )[0]
+        assert img_info['username'] == new_username
+
+        # Delete Image
+        result = ComputeResource.image_delete(
+            {'name': new_img_name, 'compute-resource': module_azurerm_cr.name}
+        )[0]
+        assert result['message'] == 'Image deleted.'
+        assert result['name'] == new_img_name
+
+    @tier2
+    def test_positive_check_available_networks(self, azurermclient, module_azurerm_cr):
+        """Check networks from AzureRm CR are available to select during host provision.
+
+        :id: 9e08463c-c700-47fc-8a58-e03aa8bcd097
+
+        :expectedresults: All the networks from AzureRM CR should be available.
+
+        :CaseLevel: Integration
+        """
+
+        result = ComputeResource.networks({'id': module_azurerm_cr.id})
+        assert len(result) > 0
```

Comment:
I wonder if this shouldn't be rather a prerequisite step in provisioning tests rather than a separate test case. If those networks are not available the provisioning will fail, or?

Reply:
no this should not be prerequisite , because this test is checking list of networks however if this fails then it does not impacting host provisioning as both are two different things ..

But for user experience for easy read it's good to see the list of n/ws and could use during provisioning ..